### DISCUSSION TITLE
Tampermonkey script for Streamlink CLI?

### DISCUSSION THREAD

**Discussion by phxn** [streamlink/streamlink#4842] [General]
I'm not asking you Streamlink devs to make it, just wondering whether something like this is possible.

A Tampermonkey script which opens stream URLs via Streamlink CLI?

I was told it was not possible, but I don't see why it wouldn't be? The script would just need to dynamically change "https://twitch.tv/<streamer>" links and extract the streamers name to send "streamlink twitch.tv/<streamer>" locally.

**Comment by bastimeyer** [comment]
I don't use any browser extensions like this, but you'd need to be able to launch and execute applications on your host system from your browser's web extensions API, and I don't think this is possible (similar to NodeJS's `child_process` module).
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions
- https://nodejs.org/api/child_process.html

A different solution is hosting a local HTTP server which can react to HTTP requests made by a browser extension (or any other HTTP client) and which then launches Streamlink for you on your host system on each valid/authorized HTTP request. See https://streamlink.github.io/thirdparty.html#liveproxy for a Streamlink wrapper which does exactly that.

**Comment by back-to** [comment]
> A Tampermonkey script which opens stream URLs via Streamlink CLI?

There is a **Firefox addon**,
you can open any URL from the browser with just a click.

https://addons.mozilla.org/firefox/addon/open-with/
https://github.com/darktrojan/openwith

old guide
https://gist.github.com/back-to/191f522011deb19b0ab2d70aa5f52c12

**Comment by phxn** [comment]
Thank you both for the replies. I am just looking for a way to easily open streams with Streamlink (without Streamlink GUI).

I considered adding the m3u playlist link as a favourite in MPC-HC but I am guessing that link changes every time someone goes live again.

I could always create .bat files to launch streams but it's still not as sleek as I'd like.
